src/core/SafeOp.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.20;
+
+import {
+    PackedUserOperation,
+    UserOperationLib
+} from "@ERC4337/account-abstraction/contracts/core/UserOperationLib.sol";
+import { SAFE_OP_TYPEHASH, ISafeOp } from "../interfaces/ISafeOp.sol";
+
+abstract contract SafeOp is ISafeOp {
```

Comment:
moved this logic to an abstract contract as this functionality has to be utilized by both Safe7579 and Launchpad